hack/e2e/kops.sh

```diff
@@ -88,36 +93,63 @@ function kops_delete_cluster() {
   ${BIN} delete cluster --name "${CLUSTER_NAME}" --state "${KOPS_STATE_FILE}" --yes
 }
 
-# TODO switch this to python or work exclusively with yaml, all this
-# hacking with jq stinks!
+# TODO switch this to python, work exclusively with yaml, use kops toolbox
+# template/kops set?, all this hacking with jq stinks!
 function kops_patch_cluster_file() {
-  CLUSTER_FILE=${1}    # input must be json
+  CLUSTER_FILE=${1}    # input must be yaml
```

Comment:
This is a function I'd like to see done in something other than bash.  Good candidate for moving reusable parts of this to the test-infra repository?  (Not PR blocking, but something we can consider for the future).

Reply:
yes, see the comment. 

I know I have gone back and forth on this a lot, I decided we should use Go. You can see here https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/927/files#diff-1174eb3051f9c1a9e1f65c760347ef47446d5b93c712925b446b29ff591184e8R36 I'm trying it out and it's fine.